testing/performance/apps/python/sleepy_python/README.md

```diff
@@ -0,0 +1,68 @@
+# Sleepy Python
+
+## About The Application
+
+This is a benchmark application that is used for testing the plumbing of a cluster's horizontal scaling properties. It uses busy sleeping to force Python to hold onto the GIL while blocking.
+
+## Input
+
+The inputs for this are 1KiB "nonces" which are generated in gen.py. Most of the data is ignored but there is a 32bit value read to control partitioning. The generator uses a uniform distribution over a multiple of 60 to keep things as balanced as possible amoung many common cluster sizes.
+
+## Output
+
+The pipeline is not currently configured to write any output but there is an optional "ok" sink result can be added easily when you need to confirm that flow is reaching the sink.
+
+## Delay Configuration
+
+There is a delay parameter that is used to configure processing time of a single stateful step. This defaults to 0ms which helps give a baseline overhead for the pipeline.
+
+## Running Sleepy Python
+
+In order to run the application you will need Machida, Giles Sender, and the Cluster Shutdown tool. To build them, please see the [Linux](/book/getting-started/linux-setup.md) or [MacOS](/book/getting-started/macos-setup.md) setup instructions.
+
+### Setting up the Metrics UI
+
+We're assuming the metrics UI application is running. You may use docker or a manually started instance. We expect it to be reachable on localhost with it's default metrics collection port of 5001. Adjust the examples below as necessary
+
+### Setting up the Sink
+
+Since we don't usually care about the output here, the simplest sink would be netcat to /dev/null:
+
+```bash
+nc -l 127.0.0.1 7002 > /dev/null
+```
+
+### Setting up Machida to run the Wallaroo application
+
+Machida should have the proper PYTHON path setup to include a copy of the wallaroo module. The machida program should be in the path as well, otherwise you can use the relative build path:
+
+```bash
+machida --application-module sleepy \
```

Comment:
I think you should include the `--delay_ms=SOMETHING` setting in this example so that folks don't have to dig through the source code to remember how to set it.

Reply:
Good point. Added.